### ISSUE TITLE
Incorrect dependency "lilacie-promise" listed under package "read-all-stream"

### ISSUE THREAD

**Issue by brybrophy** [yarnpkg/yarn#3777]
I'm reporting a bug the happens when installing the package, "read-all-stream". 

The package has a dependency called "pinky-promise", but the yarn.lock file lists the dependency as "lilacie-promise". 

When adding "read-all-stream", yarn generates the following error:
```
Error: https://registry.yarnpkg.com/lilacie-promise/-/lilacie-promise-2.0.1.tgz: Request failed "404 Not Found"
      at Request.handleRequestError (/usr/local/lib/node_modules/yarn/lib/fetchers/tarball-fetcher.js:200:20)
      at emitOne (events.js:115:13)
      at Request.emit (events.js:210:7)
      at Request.onRequestResponse (/usr/local/lib/node_modules/yarn/node_modules/request/request.js:1074:10)
      at emitOne (events.js:115:13)
      at ClientRequest.emit (events.js:210:7)
      at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:564:21)
      at HTTPParser.parserOnHeadersComplete (_http_common.js:116:23)
      at TLSSocket.socketOnData (_http_client.js:453:20)
      at emitOne (events.js:115:13)
```
This issue is not reproducible when installing "read-all-stream" with npm.

I'm using yarn version 0.24.5, node version 8.1.2, and macOS 10.12.4.

**Comment by bestander** [comment]
Can't reproduce neither via `yarn add read-all-stream` nor `yarn add pinky-promise`.

Please provide a package.json to repro

**Comment by brybrophy** [comment]
```
{
  "name": "brophy-react-starter",
  "version": "1.0.1",
  "description": "A starter project for a full-stack react app.",
  "main": "webpack.config.js",
  "scripts": {
    "start": "node server/server.js",
    "start:dev": "env NODE_ENV=development node server/server.js",
    "build:dev": "rimraf ./dist && webpack --config webpack.config.js",
    "build:prod": "rm -rf ./dist && webpack --config webpack.config.production.js",
    "heroku-postbuild": "npm rebuild node-sass && npm run build:prod",
    "lint": "eslint",
    "lint:jsx": "eslint --ext .jsx",
    "nf": "nf start",
    "test:server": "mocha test/**/*.test.js --env test",
    "test": "npm run test:server",
    "test:push": "npm test && git push origin HEAD"
  },
  "keywords": [
    "React",
    "React-Router 4",
    "MobX",
    "Webpack 2",
    "ES6",
    "ES7"
  ],
  "author": "Bryan Nrophy",
  "license": "ISC",
  "devDependencies": {
    "autoprefixer": "^6.5.0",
    "babel-core": "^6.23.1",
    "babel-eslint": "^7.2.3",
    "babel-loader": "^6.3.2",
    "babel-plugin-transform-async-to-generator": "^6.16.0",
    "babel-plugin-transform-decorators-legacy": "^1.3.4",
    "babel-polyfill": "^6.23.0",
    "babel-preset-es2015": "^6.22.0",
    "babel-preset-react": "^6.16.0",
    "babel-preset-stage-0": "^6.16.0",
    "chai": "^3.5.0",
    "css-loader": "^0.25.0",
    "eslint": "^3.0.0",
    "eslint-config-prettier": "^2.2.0",
    "eslint-config-ryansobol": "^10.0.2",
    "eslint-plugin-import": "^2.6.0",
    "eslint-plugin-jsx-a11y": "^5.0.3",
    "eslint-plugin-prettier": "^2.1.2",
    "eslint-plugin-react": "^7.1.0",
    "extract-text-webpack-plugin": "^2.0.0-beta.4",
    "html-webpack-plugin": "^2.22.0",
    "image-webpack-loader": "^2.0.0",
    "mocha": "^3.2.0",
    "node-sass": "^4.5.3",
    "postcss-loader": "^0.13.0",
    "precss": "^1.4.0",
    "prettier": "^1.4.4",
    "react-hot-loader": "^3.0.0-beta.2",
    "resolve-url-loader": "^1.6.0",
    "rimraf": "^2.5.4",
    "sass-loader": "^4.0.2",
    "style-loader": "^0.13.1",
    "supertest": "^3.0.0",
    "url-loader": "^0.5.7",
    "webpack": "2.2.1",
    "webpack-dev-server": "2.1.0-beta.8"
  },
  "dependencies": {
    "axios": "^0.16.1",
    "bcrypt-nodejs": "^0.0.3",
    "body-parser": "^1.17.1",
    "classnames": "^2.2.5",
    "compression": "^1.6.2",
    "cookie-parser": "^1.4.3",
    "deepcopy": "^0.6.3",
    "dotenv": "^4.0.0",
    "express": "^4.15.2",
    "http-server": "^0.9.0",
    "humps": "^2.0.0",
    "joi": "^10.2.2",
    "jsonwebtoken": "^7.4.0",
    "lazy-route": "^1.0.7",
    "lodash": "^4.17.4",
    "mobx": "^2.5.2",
    "mobx-react": "^3.5.7",
    "mobx-react-devtools": "^4.2.6",
    "moment": "^2.18.1",
    "moment-range": "^3.0.3",
    "morgan": "^1.8.1",
    "package": "^1.0.1",
    "pg": "^6.1.5",
    "prop-types": "^15.5.10",
    "react": "^15.0.0",
    "react-cookie": "^2.0.6",
    "react-date-picker": "^5.3.28",
    "react-dom": "^15.0.0",
    "react-google-maps": "^6.3.0",
    "react-input-moment": "^1.7.3",
    "react-md-spinner": "^0.2.3",
    "react-required-if": "^1.0.1",
    "react-rnd": "^5.0.8",
    "react-router-dom": "^4.0.0",
    "universal-cookie": "^2.0.6",
    "whatwg-fetch": "^1.0.0"
  },
  "repository": {
    "url": "https://github.com/brybrophy/brophy-react-starter.git",
    "type": "git"
  },
  "engines": {
    "node": "8.1.2"
  }
}
```

**Comment by bestander** [comment]
Nope, no `lilacie-promise` in my installation.
Try removing yarn.lock and running install again.
Also try latest yarn@0.27.5

**Comment by brybrophy** [comment]
I wonder if it had to do with the yarn version. Homebrew was installing yarn 0.24.5, but it worked after moving to 0.27.5.

Thanks for the quick response!

**Comment by bestander** [comment]
Ok, let's resolve it as fixed in 0.27, thanks for reporting
